src/py/flwr/superexec/exec_servicer.py

```diff
@@ -63,25 +64,35 @@ def StartRun(
         """Create run ID."""
         log(INFO, "ExecServicer.StartRun")
 
-        run = self.executor.start_run(
-            request.fab.content,
-            user_config_from_proto(request.override_config),
-            user_config_from_proto(request.federation_config),
-        )
+        if isinstance(self.executor, DeploymentEngine):
+            run_id = self.executor.register_run(
+                request.fab.content,
+                user_config_from_proto(request.override_config),
+            )
+            if run_id is None:
+                log(ERROR, "Executor failed to start run")
+                return StartRunResponse()
+        else:
+            run = self.executor.start_run(
+                request.fab.content,
+                user_config_from_proto(request.override_config),
+                user_config_from_proto(request.federation_config),
+            )
```

Comment:
Is this an ad-hoc solution?

Reply:
@panh99 i realised indeed introducing this isn't strictly needed to make `flwr-serverapp` functional. I've limited the changes to simply disable the execution of `flower-server-app` in the deployment's `start_run`.